### PR TITLE
drop support for LLVM <6.0.0 throughout Julia

### PR DESCRIPTION
**Pull request by vchuravy**
Post 1.0 spring cleaning!

### CODE REVIEW

File: src/disasm.cpp [hunk 1]
```diff
@@ -683,15 +673,6 @@ static void jl_dump_asm_internal(
     MCAsmBackend *MAB = 0;
     if (ShowEncoding) {
         CE = TheTarget->createMCCodeEmitter(*MCII, *MRI, Ctx);
-#if JL_LLVM_VERSION >= 60000
-        MCTargetOptions Options;
-        MAB = TheTarget->createMCAsmBackend(*STI, *MRI, Options);
```
Comment (KristofferC) [inline src/disasm.cpp]: Should this be kept?

Comment (Keno) [reply src/disasm.cpp]: yes

**Review by Keno — APPROVED** [review]

**Comment by vchuravy** [comment]
Will merge this in the morning, absent any objections.

**Comment by StefanKarpinski** [comment]
Oh man, removing 28k LOC is pretty sweet.

**Comment by vchuravy** [comment]
It's only 600ish without the patches.

**Pull request merged**